core/src/main/java/com/sequenceiq/distrox/v1/distrox/service/DistroXService.java

```diff
@@ -49,11 +53,15 @@ public StackV4Response post(DistroXV1Request request) {
     private void validate(DistroXV1Request request) {
         DetailedEnvironmentResponse environment = Optional.ofNullable(environmentClientService.getByName(request.getEnvironmentName()))
                 .orElseThrow(() -> new BadRequestException("No environment name provided hence unable to obtain some important data"));
-        if (environment != null && environment.getEnvironmentStatus() != EnvironmentStatus.AVAILABLE) {
-            throw new BadRequestException(String.format("Environment state is %s instead of AVAILABLE", environment.getEnvironmentStatus()));
+        if (environment == null) {
+            throw new BadRequestException(String.format("Cannot fetch the environment by name: %s", request.getEnvironmentName()));
+        }
+        DescribeFreeIpaResponse freeipa = freeipaClientService.getByEnvironmentCrn(environment.getCrn());
+        if (!freeipa.getStatus().isAvailable()) {
+            throw new BadRequestException(String.format("Freeipa state is %s instead of AVAILABLE", freeipa.getStatus()));
```

Comment:
If you want to provision a Data Hub then the FreeIPA instance must be running in the '%s' Environment.